src/Microsoft.Azure.SignalR.Common/Auth/MicrosoftEntra/AccessKeySynchronizer.cs

```diff
@@ -34,32 +35,50 @@ internal AccessKeySynchronizer(ILoggerFactory loggerFactory, bool start)
         {
             _ = UpdateAccessKeyAsync();
         }
-        _factory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+        _logger = (loggerFactory ?? NullLoggerFactory.Instance).CreateLogger<AccessKeySynchronizer>();
     }
 
-    public void AddServiceEndpoint(ServiceEndpoint endpoint)
+    public Task AddServiceEndpoint(ServiceEndpoint endpoint)
     {
         if (endpoint.AccessKey is MicrosoftEntraAccessKey key)
         {
-            _ = key.UpdateAccessKeyAsync();
+            _keyMap.TryAdd(key, true);
```

Comment:
how about storing the the Task (InitializeAccessKeyTask(key)) as the value, so that you could track the task status, if needed